errors_test.go

```diff
@@ -118,3 +118,39 @@ func TestGetType(t *testing.T) {
 	}
 
 }
+
+func TestGetDetails(t *testing.T) {
+	details := struct{ foo string }{"bar "}
+	tests := []struct {
+		err      error
+		expected []interface{}
+	}{
+		{nil, []interface{}{}},
+		{io.EOF, []interface{}{}},
+		{DetailsAdd(io.EOF, details), []interface{}{details}},
+		{NoType.Details(details), []interface{}{details}},
+		{DetailsAdd(BadRequest.Details(details), details), []interface{}{details, details}},
+		{DetailsAdd(DetailsAdd(DetailsAdd(io.EOF, details), details), details), []interface{}{details, details, details}},
```

Comment:
try to add a use case with different types of details and see they don't override

Reply:
Done.